websites/M/Miruro/presence.ts

```diff
@@ -0,0 +1,120 @@
+const presence = new Presence({
+		clientId: "1280494760418738247",
+	}),
+	browsingTimestamp = Math.floor(Date.now() / 1000);
+
+const enum Assets {
+	Logo = "https://i.imgur.com/Wwd5S5f.png",
+}
+
+presence.on("UpdateData", async () => {
+	const { pathname, href } = document.location,
+		presenceData: PresenceData = {
+			name: "Miruro",
+			details: "Browsing",
+			largeImageKey: Assets.Logo,
+			startTimestamp: browsingTimestamp,
+			type: ActivityType.Watching,
+		};
+
+	if (pathname.startsWith("/info")) {
+		const infoTitle = document
+				.querySelector<HTMLTitleElement>("title")
+				?.textContent.trim(),
+			infoCover = document.querySelector<HTMLImageElement>(
+				"img[src*='/file/anilistcdn/media/anime/cover/large']"
+			)?.src;
+
+		if (infoTitle) presenceData.details = infoTitle;
+		presenceData.state = "Diving into Details";
+		if (infoCover) {
+			presenceData.largeImageKey = infoCover;
+			presenceData.smallImageKey = Assets.Reading;
+			presenceData.smallImageText = "Reading info";
+		}
+	} else if (pathname.startsWith("/watch")) {
+		const video = document.querySelector<HTMLVideoElement>("video"),
+			currentAnimeTitle =
+				document.querySelector<HTMLElement>(".anime-title a")?.textContent,
+			episodeNumber = document
+				.querySelector<HTMLElement>(".ep-number")
+				?.textContent?.match(/\d+/)?.[0],
+			currentAnimeCover = document.querySelector<HTMLImageElement>(
+				"img[src*='/file/anilistcdn/media/anime/cover/large']"
+			)?.src;
+
+		if (currentAnimeTitle) {
+			presenceData.details = currentAnimeTitle;
+			if (episodeNumber) presenceData.state = `Episode ${episodeNumber}`;
+			presenceData.buttons = [{ label: "Watch Along", url: href }];
+		}
+		if (currentAnimeCover) presenceData.largeImageKey = currentAnimeCover;
+
+		if (video) {
+			if (!video.paused) {
+				[presenceData.startTimestamp, presenceData.endTimestamp] =
+					presence.getTimestampsfromMedia(video);
+				presenceData.smallImageKey = Assets.Play;
+				presenceData.smallImageText = "Now Watching";
+			} else {
+				presenceData.smallImageKey = Assets.Pause;
+				presenceData.smallImageText = "Paused";
+			}
+		} else presenceData.smallImageKey = Assets.Logo;
+	} else if (pathname.startsWith("/search")) {
+		const searchQuery = new URLSearchParams(window.location.search).get(
```

Comment:
Use `document.location` for repository consistency
```suggestion
		const searchQuery = new URLSearchParams(document.location.search).get(
```